mne/preprocessing/ieeg/_projection.py

```diff
@@ -106,3 +107,92 @@ def project_sensors_onto_brain(info, trans, subject, subjects_dir=None,
+                                   copy=True, verbose=None):
+    """Project sensors onto the brain surface.
+
+    Parameters
+    ----------
+    %(info_not_none)s
+    %(trans_not_none)s
+    %(subject)s
+    %(subjects_dir)s
+    %(picks_base)s only ``seeg`` channels.
+    %(max_dist_ieeg)s
+    flat : bool
+        Whether to project the sensors onto the flat map of the
+        inflated brain instead of the normal inflated brain.
+    copy : bool
+        If ``True``, return a new instance of ``info``, if ``False``
+        ``info`` is modified in place.
+    %(verbose)s
+
+    Returns
+    -------
+    %(info_not_none)s
+
+    Notes
+    -----
+    This is useful in sEEG analysis for visualization
+    """
+    from scipy.spatial.distance import cdist
+    _validate_type(copy, bool, 'copy')
+    if copy:
+        info = info.copy()
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    surf_data = dict(lh=dict(), rh=dict())
+    x_dir = np.array([1., 0., 0.])
+    surfs = ('pial', 'inflated')
+    if flat:
+        surfs += ('cortex.patch.flat',)
+    for hemi in ('lh', 'rh'):
+        for surf in surfs:
+            for img in ('', '.T1', '.T2', ''):
+                surf_fname = op.join(subjects_dir, subject, 'surf',
+                                     f'{hemi}.{surf}')
+                if op.isfile(surf_fname):
+                    break
+            if surf.split('.')[-1] == 'flat':
+                surf = 'flat'
+                coords, faces, orig_faces = _read_patch(surf_fname)
+                # rotate 90 degrees to get to a more standard orientation
+                # where X determines the distance between the hemis
+                coords = coords[:, [1, 0, 2]]
+                coords[:, 1] *= -1
+            else:
+                coords, faces = read_surface(surf_fname)
+            if surf in ('inflated', 'flat'):
+                x_ = coords @ x_dir
+                coords -= np.max(x_) * x_dir if hemi == 'lh' else \
+                    np.min(x_) * x_dir
+            surf_data[hemi][surf] = (coords / 1000, faces)  # mm -> m
+    # get channel locations
+    picks_idx = _picks_to_idx(info, 'seeg' if picks is None else picks)
+    locs = np.array([info['chs'][idx]['loc'][:3] for idx in picks_idx])
+    trans = _ensure_trans(trans, 'head', 'mri')
+    locs = apply_trans(trans, locs)
+    # initialize projected locs
+    proj_locs = np.zeros(locs.shape) * np.nan
+    surf = 'flat' if flat else 'inflated'
+    for hemi in ('lh', 'rh'):
+        hemi_picks = np.where(
+            locs[:, 0] <= 0 if hemi == 'lh' else locs[:, 0] > 0)[0]
+        # compute distances to pial vertices
+        dists = cdist(surf_data[hemi]['pial'][0], locs[hemi_picks])
+        for j, idx in enumerate(hemi_picks):
+            if dists[:, j].min() / 1000 < max_dist:
+                proj_locs[idx] = \
+                    surf_data[hemi][surf][0][np.argmin(dists[:, j])]
```

Comment:
We should really use `_compute_nearest` for this, it should be much faster than a naive `argmin(cdist)` (or here, two of them implicitly via a `min` then `argmin`). Let's try to remember in a follow-up PR. If it's not clear I can do it

Reply:
Yeah @agramfort brought that up too, will do, I think because the number of sensors is small, the gains are negligible but there are much larger sensor arrays coming out and definitely good to be ahead of that